src/transformers/models/fastspeech2/modeling_fastspeech2.py

```diff
@@ -0,0 +1,698 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Team The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch FastSpeech2 model."""
+
+
+import math
+from dataclasses import dataclass
+from typing import Optional, Tuple
+
+import numpy as np
+import torch
+import torch.utils.checkpoint
+from torch import nn
+
+from ...file_utils import (  # add_code_sample_docstrings,; add_start_docstrings,; add_start_docstrings_to_model_forward,; replace_return_docstrings,
+    ModelOutput,
+)
+from ...modeling_utils import (  # apply_chunking_to_forward,; find_pruneable_heads_and_indices,; prune_linear_layer,
+    PreTrainedModel,
+)
+from ...utils import logging
+from .configuration_fastspeech2 import FastSpeech2Config
+
+
+logger = logging.get_logger(__name__)
+
+_CHECKPOINT_FOR_DOC = "fastspeech2"
+_CONFIG_FOR_DOC = "FastSpeech2Config"
+_TOKENIZER_FOR_DOC = "FastSpeech2Tokenizer"
+
+FASTSPEECH2_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "fastspeech2",
+    # See all FastSpeech2 models at https://huggingface.co/models?filter=fastspeech2
+]
+
+
+def lengths_to_padding_mask(lens: torch.LongTensor) -> torch.BoolTensor:
+    # from https://github.com/pytorch/fairseq/blob/main/fairseq/data/data_utils.py
+    bsz, max_lens = lens.size(0), torch.max(lens).item()
```

Comment:
Please use proper names for variable. `lengths` is three character longer than `lens` and everyone will understand it. Likewise, use `max_lengths` and `batch_size` below.

Reply:
Thanks for pointing that out, and apologies for making you repeat yourself in the comments above and below. I've gone through a first pass through the codebase to rename ambiguous/short variables (e.g., `x`), and I'll make sure to fix any remaining ones before the next review.